front/src/App.js

```diff
@@ -37,8 +37,9 @@ function App() {
   const [cursorPosition, setCursorPosition] = useState()
   const [opponentCursorPositions, setOpponentCursorPositions] = useState({})
   const [wrongChars, setWrongChars] = useState()
-  const [messages, setMessages] = useState([])
+  const [dataMessages, setDataMessages] = useState([]) //An array contains objects with structure {sender: sender, content: content, playerID: playerID}
```

Comment:
Why rename messages?

Reply:
Because I thought at the first time, messages state only contains user message's contents, nothing else. Therefore I change to dataMessages mean it will not contain only message's content but also the player id, sender as well. If it not improve the semantic I will change back

Reply:
I see. I think leaving a comment describing it is enough for now. In the future I will possibly add type annotations with TypeScript which should make it very clear.